Packs/SailPointIdentityNow/Integrations/SailPointIdentityNow/SailPointIdentityNow.py

```diff
@@ -0,0 +1,622 @@
+        markdown = tableToMarkdown('Access Profile(s)', objects, headers=headers)
+    elif object_type == 'IdentityNow.Role':
+        headers = ['id', 'name', 'description', 'accessProfiles', 'accessProfileCount', 'created', 'modified', 'synced',
+                   'enabled', 'requestable', 'requestCommentsRequired', 'owner', 'pod', 'org', 'type']
+        markdown = tableToMarkdown('Role(s)', objects, headers=headers)
+    elif object_type == 'IdentityNow.Entitlement':
+        headers = ['id', 'name', 'displayName', 'description', 'modified', 'synced', 'source', 'privileged',
+                   'identityCount', 'attribute', 'value', 'schema', 'pod', 'org', 'type']
+        markdown = tableToMarkdown('Entitlement(s)', objects, headers=headers)
+    elif object_type == 'IdentityNow.Event':
+        headers = ['id', 'name', 'stack', 'created', 'synced', 'objects', 'ipAddress', 'technicalName', 'target',
+                   'actor', 'action', 'attributes', 'operation', 'status', 'pod', 'org', 'type']
+        markdown = tableToMarkdown('Event(s)', objects, headers=headers)
+    return markdown
+
+
+def build_results(prefix: str, key_field: str, response=None):
+    """
+    Build results.
+
+    :type prefix: ``str``
+    :param prefix: Prefix for CommandResults as part of the results.
+
+    :type key_field: ``str``
+    :param key_field: Key field for CommandResults as part of the results.
+
+    :type response: ``response``
+    :param response: Response object from IdentityNow API call.
+
+    :return: CommandResults in case of a successful response else message describing the error status.
+    """
+    if response is not None and 200 <= response.status_code < 300:
+        if isinstance(response.json(), list):
+            markdown = '### Results:\nTotal: ' + str(len(response.json())) + '\n'
+        else:
+            markdown = '### Results:\n'
+        markdown += get_markdown(prefix, response.json())
+        return CommandResults(
+            readable_output=markdown,
+            outputs_prefix=prefix,
+            outputs_key_field=key_field,
+            outputs=response.json()
+        )
+    else:
+        return None
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_connection(base_url: str, client_id: str, client_secret: str, grant_type: str):
+    """
+    Test connectivity to IdentityNow
+
+    :type base_url: ``str``
+    :param base_url: Base URL of the IdentityNow tenant.
+
+    :type client_id: ``str``
+    :param client_id: Client Id for OAuth 2.0.
+
+    :type client_secret: ``str``
+    :param client_secret: Client Secret for OAuth 2.0.
+
+    :type grant_type: ``str``
+    :param grant_type: Grant Type for OAuth 2.0. Defaulted to 'client_credentials' if not provided.
+
+    :return: HTTP connectivity status for test connection.
+    """
+    header = get_headers(base_url, client_id, client_secret, grant_type)
+    if header is not None:
+        return 'ok'
+    else:
+        return 'Unable to connect to IdentityNow!'
```

Comment:
I would suggest using 'try except' statements and return the message returned from the API for a clearer error message